cmake/compiler/gcc/target.cmake

```diff
@@ -106,6 +106,10 @@ else()
     list(APPEND TOOLCHAIN_C_FLAGS
       -mcpu=${GCC_M_CPU}
       )
+  elseif("${ARCH}" STREQUAL "riscv32")
+    list(APPEND TOOLCHAIN_C_FLAGS
+      -march=rv32imac -mabi=ilp32 -mcmodel=medlow
```

Comment:
How does this co-exist with

/home/sebo/ncs/zephyr/soc/riscv32/pulpino/CMakeLists.txt
5:  -march=IMXpulpv2

?

Also, other platforms have set arch and abi through 'soc' and 'arch' build scripts, so I believe we should move this into either arch/riscv32 or soc/riscv32/*.